### PR TITLE
feat: implemented fetch of edge version with a fallback option

### PR DESCRIPTION
**Pull request by vickywane**
This PR resolves [this](https://github.com/ambianic/ambianic-ui/issues/343)

## Purpose
With the new modification in this PR, the connected edge device version is fetched and stored in the VueX store. For backward compatibility, a fallback has been implemented for this feature, considering only updated edge devices have the edge version in the `api/status` endpoint.

## Merge Checklist
- [x]  The code change is tested and works locally.
- [x]  The user and dev documentation is up to date.
- [x]  There is no commented-out code in this PR.
- [x]  No lint errors (use flake8)
- [ ]  100% test coverage

### CODE REVIEW

**Comment by vickywane** [comment]
LGTM alert has been fixed in the second commit. I was expecting it to re-run the check.

**Comment by vickywane** [comment]
@ivelin something to note.

I had an error with the `infinite-loading` component. It kept throwing an error similar to this [issue](https://github.com/PeachScript/vue-infinite-loading/issues/55) and was breaking the lighthouse `no-browser-error` check.

Attached is a screenshot of the error in the browser.
infinite-loader-error

File: src/store/index.js [hunk 1]
```diff
@@ -5,19 +5,30 @@ import { INITIALIZE_PNP } from './action-types.js'
 import { UPDATE_AVAILABLE } from './mutation-types'
 
 Vue.use(Vuex)
+let edgeVersion = require('@/../package.json').version
+
+fetch('http://localhost:8778/api/status')
```
Comment (ivelin) [inline src/store/index.js:10]: @vickywane this function calling the status API belongs in the EdgeAPI class:

https://github.com/ambianic/ambianic-ui/blob/master/src/remote/edgeAPI.js

File: src/store/index.js [hunk 2]
```diff
@@ -5,19 +5,30 @@ import { INITIALIZE_PNP } from './action-types.js'
...
+  })
+  .catch((e) => {
+    console.log(`Error getting version: ${e}`)
+  })
 
 const store = new Vuex.Store({
   state: {
     updateToBeInstalled: undefined,
-    version: require('@/../package.json').version
+    edgeVersion
```
Comment (ivelin) [inline src/store/index.js:24]: @vickywane this is a rushed patch.

We still need to show the PWA version in the About page. Separately from it, we want to show the Edge version in the edge card under settings.

Please add at least two new test cases to prevent future regression like this change:

1. Check that PWA version is correctly displayed in the About page.
2. Check that Edge version is correctly displayed in the Edge Settings Card.

**Review by ivelin — CHANGES_REQUESTED** [review]
@vickywane several comments added.

We also need test cases to cover graceful handling of older edge versions.

Please submit for review when you have achieved 100% coverage for new code in this PR.

**Comment by ivelin** [comment]
> @ivelin something to note.
> 
> I had an error with the `infinite-loading` component. It kept throwing an error similar to this [issue](https://github.com/PeachScript/vue-infinite-loading/issues/55) and was breaking the lighthouse `no-browser-error` check.
> 
> Attached is a screenshot of the error in the browser.
> infinite-loader-error

Hmm. I have not seen this problem before. Did the `<div/>` wrapper help?

**Comment by ivelin** [comment]
> > @ivelin something to note.
> > I had an error with the `infinite-loading` component. It kept throwing an error similar to this [issue](https://github.com/PeachScript/vue-infinite-loading/issues/55) and was breaking the lighthouse `no-browser-error` check.
> > Attached is a screenshot of the error in the browser.
> > infinite-loader-error
> 
> Hmm. I have not seen this problem before. Did the `<div/>` wrapper help?

Is it possible that the timelinefetch function returns a list too long that causes the max length error?
